Update vLLM options for MLA test

yoo-kumaneko opened the pull request:
I've enabled some MoE-related options, like EP and EPLB. (Removed, EP not working)
Also, I've incorporated quantization configurations that are currently used in a production environment at Tencent. I hope this could make this test closer to real-world use cases.

In addition, because TP=2 requires two GPUs, I've added a gpu_count attribute under the docker field to the test yaml, and I've modified the GPU allocation script to support more than one GPU for the test container.

Commit:
Add a few vLLM options

Signed-off-by: rigginschen <rigginschen@tencent.com>

maobaolong commented on .buildkite/configs/local_cpu_mla.yaml:
```diff
@@ -16,4 +16,12 @@ vllm:
     - "--no-enable-prefix-caching"
     - "--kv-transfer-config"
-    - "{\"kv_connector\":\"LMCacheConnectorV1\",\"kv_role\":\"kv_both\"}"
+    - "{\"kv_connector\":\"LMCacheConnectorV1Dynamic\",\"kv_role\":\"kv_both\",\"kv_connector_module_path\":\"lmcache.integration.vllm.lmcache_connector_v1\",\"kv_connector_extra_config\":{\"discard_partial_chunks\":true}}"
```
Could you have a try with this suggestion? It look much clear.

yoo-kumaneko replied:
Sure

Commit:
remove the escape sign

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
add tp

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
fix typo bug

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
Update gpu allocation logic to use two gpus

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
fix tedious docker device assignment problem

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
remove ep for now

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
Use yaml for config rather than env var. Lower exp gain temporarily.

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
revert incorrect formatting

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
Update lmcache config yaml

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
Update config yaml

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
add logic for required gpu count

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
Put gpu count under docker field

Signed-off-by: rigginschen <rigginschen@tencent.com>

yoo-kumaneko commented:
@Shaoting-Feng Hi Shaoting, would you like to take a look at this PR? I've modified the GPU allocation script so that we can allocate more than one GPUs to a container. I also added some flags like TP=2 to the test yaml.

Shaoting-Feng commented on .buildkite/configs/lmcache_config.yaml:
```diff
@@ -0,0 +1,15 @@
+# LMCache Configuration File
+# This replaces the individual environment variables for cleaner configuration
```
Shall we rename this file to show this is for mla test?

yoo-kumaneko replied:
Yes, will do

yoo-kumaneko replied:
File renamed

Shaoting-Feng commented on .buildkite/scripts/vllm-integration-tests.sh:
```diff
@@ -111,16 +111,18 @@ run_lmcache_vllmopenai_container() {
-        --gpus "device=${best_gpu}"
+        --gpus "\"device=${best_gpu}\""
         --volume ~/.cache/huggingface:/root/.cache/huggingface
+        --volume "${CONFIG_DIR}/lmcache_config.yaml:/etc/lmcache/config.yaml:ro"
```
We can also mount the entire config directory so that it doesn’t have to be only for the MLA test.

yoo-kumaneko replied:
Got it

yoo-kumaneko replied:
I've put the config file into a config folder and mounted the folder instead.

Shaoting-Feng replied:
Maybe you forgot this part

yoo-kumaneko replied:
My bad. I forgot to git add this file

yoo-kumaneko replied:
Thanks for the sharp eyes. Now it's fixed.

Commit:
mount the config directory instead of specific files

Signed-off-by: rigginschen <rigginschen@tencent.com>

Commit:
update mount path

Signed-off-by: rigginschen <rigginschen@tencent.com>

Shaoting-Feng reviewed (approved):
Thanks! LGTM

maobaolong reviewed (approved):
LGTM, thanks for this effort!
